plugins/PostProcessingPlugin/scripts/AddCoolingProfile.py

```diff
@@ -0,0 +1,877 @@
+                    elif ";LAYER:" in fan_line:
+                        modified_data += fan_line + "\n"
+                        layer_number = str(fan_line.split(":")[1])
+                        for num in range(0,15,2):
+                            if layer_number == str(fan_list[num]):
+                                modified_data += fan_list[num + 1] + this_fan + "\n"
+                                current_fan_speed = str(fan_list[num + 1].split("S")[1])
+                                current_fan_speed = str(current_fan_speed.split(" ")[0]) #Just in case
+                    else:
+                        modified_data += fan_line + "\n"
+                if modified_data.endswith("\n"): modified_data = modified_data[0:-1]
+                data[l_index] = modified_data
+            return data
+
+        #The Single Fan "By Feature" section------------------------------
+        if by_layer_or_feature == "by_feature" and (not is_multi_fan or not is_multi_extr_print):
+            layer_number = "0"
+            index = 1
+            #Start with layer:0
+            for l_index in range(layer_0_index,len(data)-1,1):
+                modified_data = ""
+                layer = data[l_index]
+                lines = layer.split("\n")
+                for line in lines:
+                    if ";LAYER:" in line:
+                        layer_number = str(line.split(":")[1])
+                    if int(layer_number) >= int(the_start_layer) and int(layer_number) < int(the_end_layer)-1:
+                        if ";TYPE:SKIRT" in line:
+                            modified_data += fan_sp_skirt + t0_fan + "\n"
+                        elif ";TYPE:WALL-INNER" in line:
+                            modified_data += fan_sp_wall_inner + t0_fan + "\n"
+                        elif ";TYPE:WALL-OUTER" in line:
+                            modified_data += fan_sp_wall_outer + t0_fan + "\n"
+                        elif ";TYPE:FILL" in line:
+                            modified_data += fan_sp_fill + t0_fan + "\n"
+                        elif ";TYPE:SKIN" in line:
+                            modified_data += fan_sp_skin + t0_fan + "\n"
+                        elif line == ";TYPE:SUPPORT":
+                            modified_data += fan_sp_support + t0_fan + "\n"
+                        elif ";TYPE:SUPPORT-INTERFACE" in line:
+                            modified_data += fan_sp_support_interface + t0_fan + "\n"
+                        elif ";MESH:NOMESH" in line or ";MESH:NONMESH" in line:  #compatibility with 5.3.0 'NOMESH'
+                            if fan_combing == True:
+                                modified_data += "M106 S0" + t0_fan + "\n"
+                        elif ";TYPE:PRIME-TOWER" in line:
+                            modified_data += fan_sp_prime_tower + t0_fan + "\n"
+                        elif line == ";BRIDGE":
+                            modified_data += fan_sp_bridge + t0_fan + "\n"
```

Comment:
Each line of this is checking for a boolean, and then doing almost the exact same operation with one different 'parameter'. You could do something like `modified_data += mapping[";TYPE:BLAH"] + t0_fan + "\n"` instead (the mapping would need to be a function/lambda in this case instead, as the way you check for the string isn't the same here, and you'd need to check for the `fan_combing` in one particular case, but the same principle still applies).